samples/graph-meeting-notification/csharp/MeetingNotification/Bots/MeetingNotificationBot.cs

```diff
@@ -0,0 +1,109 @@
+            this.conversationReferences.AddOrUpdate(meetingUrl, conversationReference, (key, newValue) => conversationReference);
+        }
+
+        protected override async Task OnInstallationUpdateAddAsync(ITurnContext<IInstallationUpdateActivity> turnContext, CancellationToken cancellationToken)
+        {
+            try
+            {
+                var meetingInfo = await TeamsInfo.GetMeetingInfoAsync(turnContext, null, cancellationToken);
+                var decodedMeetingJoinUrl = HttpUtility.UrlDecode(meetingInfo.Details.JoinUrl.ToString());
+                AddConversationReference(turnContext.Activity as Activity, decodedMeetingJoinUrl);
+                await this.subscriptionManager.InitializeAllSubscription(meetingInfo.Details.JoinUrl.ToString());
+
+                await turnContext.SendActivityAsync(MessageFactory.Attachment(NotificationCardHelper.GetSubscriptionInitializedCard(true)), cancellationToken);
+
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    await Task.Delay(TimeSpan.FromMinutes(15), cancellationToken).ConfigureAwait(false);
+                    await this.subscriptionManager.CheckSubscriptions().ConfigureAwait(false); ;
+                }
+            }
+            catch (Exception ex)
+            {
+                this.logger.LogWarning(ex.Message);
+
+                if (ex.Message == "The meetingId can only be null if turnContext is within the scope of a MS Teams Meeting.")
+                {
+                    await turnContext.SendActivityAsync(MessageFactory.Text("Please install this bot in meeting chat to get meeting notifications"), cancellationToken);
+                }
+                else
+                {
+                    await turnContext.SendActivityAsync(MessageFactory.Text("Something went wrong! Please try installing bot again"), cancellationToken);
+                }
+                
+            }
+        }
+
+        protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+        {
+            try
+            {
+                var meetingInfo = await TeamsInfo.GetMeetingInfoAsync(turnContext, null, cancellationToken);
+                var decodedMeetingJoinUrl = HttpUtility.UrlDecode(meetingInfo.Details.JoinUrl.ToString());
+
+                AddConversationReference(turnContext.Activity as Activity, decodedMeetingJoinUrl);
+                await turnContext.SendActivityAsync(MessageFactory.Attachment(NotificationCardHelper.GetSubscriptionInitializedCard(false)), cancellationToken);
+                await this.subscriptionManager.InitializeAllSubscription(meetingInfo.Details.JoinUrl.ToString());
+            }
+            catch (Exception ex)
+            {
+                if (ex.Message == "The meetingId can only be null if turnContext is within the scope of a MS Teams Meeting.")
+                {
```

Comment:
This comparison may not the correct way to compare.
error message may change and it can fail the condition.
Can't we check the exception type or code ?

Reply:
I have added another `catch block of InvalidOperationException` to handle this.